subcommands.go

```diff
@@ -114,6 +132,46 @@ func (cdr *Commander) ImportantFlag(name string) {
 	cdr.important = append(cdr.important, name)
 }
 
+// VisitGroups visits each command group in lexicographical order, calling
+// fn for each.
+func (cdr *Commander) VisitGroups(fn func(*CommandGroup)) {
+	sort.Sort(byGroupName(cdr.commands))
+	for _, g := range cdr.commands {
+		fn(g)
+	}
+}
+
+// VisitCommands visits each command command in registered order grouped by
+// group in lexicographical order, calling fn for each.
+func (cdr *Commander) VisitCommands(fn func(*CommandGroup, Command)) {
```

Comment:
Specific reason to pass in the CommandGroup along with Command? Is it for the Name? Would it be better to just pass in the name by itself?

Reply:
Yeah, I imagine it would be more future proof this way if there are any legitimate fields that get added to the group. Visit* methods could possibly also be used for other things besides help too.